packages/core/core-flows/src/payment-collection/workflows/create-payment-session.ts

```diff
@@ -68,16 +80,89 @@ export const createPaymentSessionsWorkflow = createWorkflow(
       fields: ["id", "amount", "currency_code", "payment_sessions.*"],
       variables: { id: input.payment_collection_id },
       list: false,
+    }).config({ name: "get-payment-collection" })
+
+    const { paymentCustomer, accountHolder } = when(
+      "customer-id-exists",
+      { input },
+      (data) => {
+        return !!data.input.customer_id
+      }
+    ).then(() => {
+      const customer: CustomerDTO & { account_holder: AccountHolderDTO } =
+        useRemoteQueryStep({
+          entry_point: "customer",
+          fields: [
+            "id",
+            "email",
+            "company_name",
+            "first_name",
+            "last_name",
+            "phone",
+            "addresses.*",
+            "account_holder.*",
+            "metadata",
+          ],
+          variables: { id: input.customer_id },
+          list: false,
+        }).config({ name: "get-customer" })
+
+      const paymentCustomer = transform({ customer }, (data) => {
+        return {
+          ...data.customer,
+          billing_address:
+            data.customer.addresses?.find((a) => a.is_default_billing) ??
+            data.customer.addresses?.[0],
+        }
+      })
+
+      const accountHolderInput = {
```

Comment:
question: Can we construct data like this? I thought we always needed to use a `transform` to ensure the value of properties are resolved correctly at runtime and not "composition time".

Reply:
If you are not doing spread, ternary operators, etc. you can yes. In many places we construct the object we pass to it, here we just assign it to a variable.